core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer_v1.go

```diff
@@ -105,7 +106,11 @@ func (b *logBuffer) Enqueue(uid *big.Int, logs ...logpoller.Log) (int, int) {
 		b.setUpkeepQueue(uid, buf)
 	}
 
-	latestLogBlock, uniqueBlocks := blockStatistics(logs...)
+	latestLogBlock, uniqueBlocks, reorgBlocks := b.blockStatistics(logs...)
+	if len(reorgBlocks) > 0 {
+		b.evictReorgdLogs(reorgBlocks)
+	}
+
```

Comment:
for line 117: nit. we can lof a warning instead of debug as we don't expect that condition to happen (logs fetched from log poller should be for newer logs than previous enqueues)

Reply:
Switched this to a warn; not sure about the value of debug logs in trackBlockNumbersForUpkeep; if we don't need them I think we can just drop this function completely